src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+
+      case 'clear':
+        currentState = {};
+        break;
+
+      default:
+        throw new Error(`Unknown action type: ${action.type}`);
+    }
+    result.push({ ...currentState });
+  }
+
+  return result;
 }
```

Comment:
Consider using a more descriptive name for the object copy. Instead of `currentState`, a name like `stateClone` or `stateCopy` would be more indicative of its purpose.

Reply:
Consider will be better if you fix it 
<img width="869" alt="image" src="https://github.com/user-attachments/assets/d42e489e-48ef-4e3c-9367-2f8add7175b1">